src/main/java/com/playmonumenta/networkrelay/RabbitMQManager.java

```diff
@@ -252,26 +333,42 @@ protected void sendNetworkMessage(String destination, String channel, JsonObject
 	}
 
 	protected void sendNetworkMessage(String destination, String channel, JsonObject data, AMQP.BasicProperties properties) throws Exception {
-		JsonObject json = new JsonObject();
-		json.addProperty("source", mShardName);
-		json.addProperty("dest", destination);
-		json.addProperty("channel", channel);
-		json.add("data", data);
+		JsonObject root = new JsonObject();
+		root.add("data", data);
+		sendNetworkMessageInternal(destination, channel, root, properties);
+	}
+
+	/* Called after adding the data and (optionally) pluginData to a root json object */
+	private void sendNetworkMessageInternal(String destination, String channel, JsonObject root, AMQP.BasicProperties properties) throws Exception {
+		root.addProperty("source", mShardName);
+		root.addProperty("dest", destination);
+		root.addProperty("channel", channel);
+
+		/* Broadcasting a non-heartbeat message - add heartbeat data to it if it's been more than half the normal heartbeat time since the last heartbeat */
+		/* Note that heartbeats also go through this same method, so need to not add the same data to them twice */
```

Comment:
> Note that heartbeats also go through this same method, so need to not add the same data to them twice

It might make more sense to make the send heartbeat method send a packet *without* the heartbeat data and let this code add it automatically, since that method should send a heartbeat if more than enough time has passed to add it.

Reply:
Going to let this be as is. Agreed that refactoring might make sense but this has sat so long I just want the feature at this point.